src/bundle/Controller/ContentTypeController.php

```diff
@@ -759,6 +783,109 @@ protected function createDeleteForm(ContentTypeGroup $group, ContentType $conten
+        ?Language $language = null
+    ): void {
+        $metaFieldTypeIdentifiers = $this->contentTypeFieldTypesResolver->getMetaFieldTypeIdentifiers();
+
+        if (null === $language) {
+            $language = $this->languageService->loadLanguage($this->languageService->getDefaultLanguageCode());
+        }
+
+        foreach ($metaFieldTypeIdentifiers as $metaFieldTypeIdentifier) {
+            $fieldGroup = $this->getDefaultMetaDataFieldTypeGroup() ?? $this->fieldsGroupsList->getDefaultGroup();
+
+            if ($this->isMetaFieldDefinitionExists($metaFieldTypeIdentifier, $fieldGroup, $contentType)) {
+                continue;
+            }
+
+            $fieldDefinitionCreateStruct = $this->createMetaFieldDefinitionCreateStruct(
+                $metaFieldTypeIdentifier,
+                $fieldGroup,
+                $language,
+                $this->getNextFieldPosition($contentType)
+            );
+
+            if ($contentType instanceof ContentTypeDraft) {
+                $this->contentTypeService->addFieldDefinition($contentType, $fieldDefinitionCreateStruct);
+            }
+
+            if ($contentType instanceof ContentTypeCreateStruct) {
+                $contentType->addFieldDefinition($fieldDefinitionCreateStruct);
+            }
+        }
+    }
+
+    private function createMetaFieldDefinitionCreateStruct(
+        string $identifier,
+        string $fieldGroup,
+        Language $language,
+        int $position
+    ): FieldDefinitionCreateStruct {
+        $fieldDefinitionCreateStruct = $this->contentTypeService->newFieldDefinitionCreateStruct(
+            uniqid('field_'),
+            $identifier
+        );
+
+        $fieldDefinitionCreateStruct->fieldGroup = $fieldGroup;
+        $label = $this->translator->trans(/** @Ignore */
+            $identifier . '.name',
+            [],
+            'fieldtypes',
+            $this->localeConverter->convertToPOSIX($language->languageCode)
+        );
+
+        $fieldDefinitionCreateStruct->names = [
+            $language->languageCode => $label,
+        ];
+
+        $fieldDefinitionCreateStruct->position = $position;
+
+        return $fieldDefinitionCreateStruct;
+    }
+
+    private function isMetaFieldDefinitionExists(
+        string $fieldTypeIdentifier,
+        string $fieldTypeGroup,
+        ValueObject $contentType
+    ): bool {
+        foreach ($contentType->fieldDefinitions as $fieldDefinition) {
+            if (
+                $fieldDefinition->fieldTypeIdentifier === $fieldTypeIdentifier
+                && $fieldDefinition->fieldGroup === $fieldTypeGroup
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function getNextFieldPosition(ValueObject $contentType): int
+    {
+        $fieldDefinitions = $contentType->fieldDefinitions;
+
+        if (is_array($fieldDefinitions) && !empty($fieldDefinitions)) {
+            return end($fieldDefinitions)->position + 1;
```

Comment:
Wouldn't be possible for `end($fieldDefinitions)` to return `false` if there are no definitions?

Reply:
If there are no definitions we should return 0 instead of false because we need to know on which position definition should be added: https://github.com/ibexa/admin-ui/pull/558/files#diff-10ebccb8c3eea70eee72c66478efcb705199cfda6fe343818fdbe81ba1d96974R149